### PR TITLE
Task/trial1 cache with updates

### PR DESCRIPTION
**Pull request by kzangeli**
New sub cache working.
Not complete, nor any of our normal test suites executed, but working with new jmeter test.

### CODE REVIEW

File: src/lib/mongoBackend/mongoSubCache.cpp [hunk 1]
```diff
@@ -283,14 +500,15 @@ static void subCacheItemUpdate(const char* tenant, BSONObj* subP)
   //
   // 03. Create CachedSubscription
   //
-  CachedSubscription* cSubP = (CachedSubscription*) calloc(1, sizeof(CachedSubscription));
+  CachedSubscription* cSubP = new CachedSubscription();
 
   if (cSubP == NULL)
   {
-    LM_E(("Runtime Error (cannot allocate memory for a cached subscription: %s)", strerror(errno)));
+    LM_X(1, ("Runtime Error (cannot allocate memory for a cached subscription: %s)", strerror(errno)));
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Maybe a "FIXME: See #1362" should be included here...

File: src/lib/mongoBackend/mongoSubCache.cpp [hunk 2]
```diff
@@ -283,14 +500,15 @@ static void subCacheItemUpdate(const char* tenant, BSONObj* subP)
...
     return;
   }
 
+  // LM_M(("Allocated CachedSubscription at %p", cSubP));
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoSubCache.cpp]: I don't paying too much attengion to LM_M (either commented or not), understanding we will clean all this in the last steps of the "cache recovery" work.

NTC

Comment (kzangeli) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Yeah, the 'tough' PR comes later ... :-)

File: src/lib/mongoBackend/mongoSubCache.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/mongoBackend/mongoSubCache.cpp]: ok, no probs

Comment (kzangeli) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Fixed in 87f8aa0

**Comment by fgalan** [comment]
LGTM

**Pull request merged**